src/pages/[platform]/build-a-backend/data/custom-business-logic/connect-bedrock/index.mdx

```diff
@@ -0,0 +1,392 @@
+
+
+## Step 2 - Define a custom query
+
+<BlockSwitcher>
+<Block name="Function">
+Next, replace the contents of your `amplify/data/resource.ts` file with the following code. This will define and export a lambda function that was granted permission to invoke a generative AI model in Amazon Bedrock in the previous step. A custom query named `generateHaiku` is added to the schema with the `generateHaikuFunction` as the handler using the `a.handler.function()` modifier:
+
+```ts title="amplify/data/resource.ts"
+import {
+  type ClientSchema,
+  a,
+  defineData,
+  defineFunction,
+} from "@aws-amplify/backend";
+
+export const generateHaikuFunction = defineFunction({
+  entry: "./generateHaiku.ts",
+});
+
+const schema = a.schema({
+  generate: a
+    .query()
+    .arguments({ prompt: a.string().required() })
+    .returns(a.string())
+    .authorization((allow) => [allow.publicApiKey()])
+    .handler(a.handler.function(generateHaikuFunction)),
+});
+
+export type Schema = ClientSchema<typeof schema>;
+
+export const data = defineData({
+  schema,
+  authorizationModes: {
+    defaultAuthorizationMode: "apiKey",
+    apiKeyAuthorizationMode: {
+      expiresInDays: 30,
+    },
+  },
+});
+```
+</Block>
+<Block name="Custom resolver powered by AppSync JavaScript resolvers">
+With Amazon Bedrock added as a data source, you can reference it in custom queries using the `a.handler.custom()` modifier which accepts the name of the data source and an entry point for your resolvers. Replace the contents of your `amplify/data/resource.ts` file with the following code to define a custom query named `generateHaiku` in the schema:
+
+```ts title="amplify/data/resource.ts"
+import { type ClientSchema, a, defineData } from "@aws-amplify/backend";
+
+const schema = a.schema({
+  generateHaiku: a
+    .query()
+    .arguments({ prompt: a.string().required() })
+    .returns(a.string())
+    .authorization((allow) => [allow.publicApiKey()])
+    .handler(
+      a.handler.custom({
+        dataSource: "BedrockDataSource",
+        entry: "./generateHaiku.js",
+      })
+    ),
+});
+
+export type Schema = ClientSchema<typeof schema>;
+
+export const data = defineData({
+  schema,
+  authorizationModes: {
+    defaultAuthorizationMode: "apiKey",
+    apiKeyAuthorizationMode: {
+      expiresInDays: 30,
+    },
+  },
+});
+```
+</Block>
+</BlockSwitcher>
+
+## Step 3 - Configure custom business logic handler code
+
+Next, create a `generateHaiku.js` file in your `amplify/data` folder and use the following code to define a custom resolver for the custom query added to your schema in the previous step:
+
+<BlockSwitcher>
+<Block name="Function">
+The following code uses the `BedrockRuntimeClient` from the `@aws-sdk/client-bedrock-runtime` package to invoke the generative AI model in Amazon Bedrock. The `handler` function takes the user prompt as an argument, invokes the model, and returns the generated haiku.
+
+```ts title="amplify/data/generateHaiku.ts"
```

Comment:
`generateHaiku.ts` file extension is inconsistent. The prior references show `.js` extension

Reply:
The function example is tagged as `ts` because the code imports a type for `Schema`.

I think I need to either adjust the text above to not specify an extension or mention .ts if using the lambda function approach and .js if using the AppSync js resolver approach

Reply:
Addressed in latest commit